### ISSUE TITLE
Support lint jars (don't recommend removing from implementation)

### ISSUE THREAD

**Issue by ColtonIdle** [autonomousapps/dependency-analysis-android-gradle-plugin#303]
I used this tool on my android project for the first time and it says that this dependency is unused https://github.com/littlerobots/rxlint but it is used BUT it's just a linter. Maybe that's why? Should I be using 

`lintChecks ("nl.littlerobots.rxlint:rxlint:1.7.6")`
instead of 
`implementation ("nl.littlerobots.rxlint:rxlint:1.7.6")`

 Curious if theres a way to prevent the false detection? Maybe this should be an issue on the library so that it specifies to use lintChecks?

**Comment by autonomousapps** [comment]
Can you try with lintChecks and let me know if that still works? 


On October 13, 2020, GitHub <notifications@github.com> wrote:
> I used this tool on my android project for the first time and it says

> that this project is unused <https://github.com/littlerobots/rxlint>

> but it is used BUT it's just a linter. Maybe that's why? Should I be

> using
>
> lintChecks ("nl.littlerobots.rxlint:rxlint:1.7.6")
>  instead of
>  implementation ("nl.littlerobots.rxlint:rxlint:1.7.6")
>
> Curious if theres a way to prevent the false detection? Maybe this
> should be an issue on the library so that it specifies to use
> lintChecks?
>

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/autonomousapps/dependency-analysis-android-gradle-

plugin/issues/303>, or unsubscribe
<https://github.com/notifications/unsubscribe-
auth/ABJG5PLPHLVS7XDK5ZOKMXDSKSBFLANCNFSM4SPIMSZA>.

**Comment by ColtonIdle** [comment]
I tried lintChecks and it did not work.

I also asked on the lint-dev google group to hopefully get an answer from Tor on whether or not this is potentially a misconfiguration from the library developer.

https://groups.google.com/g/lint-dev/c/cObreGf3nbQ

**Comment by ColtonIdle** [comment]
@autonomousapps doesn't seem like lintChecks is the solution here. Curious if there's anything reasonable for this plugin to do instead.

**Comment by autonomousapps** [comment]
I haven't considered in detail yet, but I wonder if I can detect and treat lint-jars in a similar way to how I detect service loaders or security providers. For those cases, I don't ever recommend removal because they're only used at runtime and impossible (or at least very challenging) to detect in the bytecode.

I will update this ticket title to reflect that I want to have built-in support for understanding lint jars.

**Comment by autonomousapps** [comment]
In the meantime, you can instruct the plugin to ignore this dependency. The wiki explains how.

**Comment by ColtonIdle** [comment]
Thank you! Glad there's a possibility to support this enhancement.

**Comment by autonomousapps** [comment]
In rxlint-1.7.6.aar, there are these files:
* R.txt (empty)
* AndroidManifest.xml (basically empty)
* lint.jar
* classes.jar (empty)

in `lint.jar`, we see a manifest and a bunch of class files.

in `MANIFEST.MF`, we have this content:
```
Manifest-Version: 1.0
Lint-Registry: nl.littlerobots.rxlint.RxIssueRegistry
```
so here's where I'd like to determine if an aar was meant for linting. 

I should check Timber as well for another example to see if it's consistent.

**Comment by autonomousapps** [comment]
Re-opening. It'll take at least one more PR to resolve this.

**Comment by autonomousapps** [comment]
Ok, now it's done. This will be resolved with v0.63.
